wezterm-input-types/src/lib.rs

```diff
@@ -1305,6 +1308,84 @@ impl Default for WindowDecorations {
     }
 }
 
+#[derive(Debug, FromDynamic, ToDynamic, Clone)]
+pub struct FancyWindowDecorations {
+    #[dynamic(default)]
+    pub is_left: bool,
+    #[dynamic(default)]
+    pub remove_hide_button: bool,
+    #[dynamic(default)]
+    pub remove_maximize_button: bool,
```

Comment:
What do you think about changing this config struct and breaking it apart into three things?

```lua
integrated_title_buttons = "HIDE|MAXIMIZE|CLOSE" 
```

OR: allow the user to control the order: 

```lua
integrated_title_buttons = {"Hide", "Maximize", "Close"}
```

Alignment as a separate option:

```lua
integrated_title_button_alignment = "Left" -- or "Right"
```

And a separate option for the style:

```lua
integrated_title_button_style = "Windows"
```

Rationale: these are a bit more flexible for the user, and there is a gotcha with default-initializing fields of structs in the lua config: when you omit fields, the `dynamic(default)` attribute takes the natural Default impl from the type of the field, but when you have struct-level Defaults for the containing struct, those are not used.  That can lead to discrepancies in the actual value used when the user wants to override just a single field.
Breaking things out makes it easier for the user, and makes it easier for a future online config editing UI to edit the individual values.

Reply:
)
![image](https://user-images.githubusercontent.com/63008755/202501959-dda12520-ffe5-46f0-bc7b-eeb28f162dea.png)

We'll be able to add other button types like that chrome's strange thing
![image](https://user-images.githubusercontent.com/63008755/202503283-cc98a0de-a786-4271-b9da-91aad552f834.png)


but this doesn't work yet for not fancy tabbar, because it reserves place only for three buttons
![image](https://user-images.githubusercontent.com/63008755/202502538-673a595d-c1cf-4a02-a9c7-c387beaf186e.png)
![image](https://user-images.githubusercontent.com/63008755/202502766-fc44af9b-ca05-4854-88a8-cdc6e063a9b8.png)

Reply:
BTW what do you think about case insentive values? Would you like to make that behavior by default for enums?
![image](https://user-images.githubusercontent.com/63008755/202510307-9bcdc773-6df3-46e0-935e-3dad123ed970.png)